src/app/_components/BlockList/data/useBlocksPageBlockListUngrouped.ts

```diff
@@ -0,0 +1,104 @@
+'use client';
+
+import { useCallback, useEffect, useRef } from 'react';
+
+import { useBlockListContext } from '../BlockListContext';
+import { useBlockListWebSocket2 } from '../Sockets/useBlockListWebSocket2';
+import { BlockListData, generateBlockList, mergeBlockLists, waitForFadeAnimation } from '../utils';
+import { useBlocksPageBlockListUngroupedInitialBlockList } from './useBlocksPageBlockListUngroupedInitialBlockList';
+
+export function useBlocksPageBlockListUngrouped(btcBlockLimit: number = 3) {
+  const { setBlockListLoading, liveUpdates } = useBlockListContext();
+
+  const {
+    initialStxBlocksHashes,
+    initialBlockList,
+    refetchInitialBlockList,
+    fetchNextPage,
+    isFetchingNextPage,
+    hasNextPage,
+  } = useBlocksPageBlockListUngroupedInitialBlockList(btcBlockLimit);
+
+  // initially the block list is the initial blocklist
+  const blockList = useRef<BlockListData[]>(initialBlockList);
+  // when the initial block list changes, reset the block list to the initial blocklist
+  //TODO: this is the issue!!!!!! it keeps firing, resetting the list back to the initial list
```

Comment:
I fixed this. Forgot to remove this comment. The root cause was useQueries used in useBlocksPageBlockListUngroupedInitialBlockList always returned a new var, and I was memoizing it, but because the var was always new, the memoizing didn't do much. The workaround was using the combine property in the useQueries config